src/mixer/audio.ts

```diff
@@ -135,19 +136,44 @@ class Player extends ((PlayerEmitter as unknown) as { new (): EventEmitter }) {
     this._applyVolume();
   }
 
+  setOutputDevice(sinkId: string) {
+    if (!this.customOutput) {
+      throw Error(
+        "Can't set sinkId when player is not in customOutput mode. Please reinit player."
+      );
+    }
+    try {
+      (this.wavesurfer as any).setSinkId(sinkId);
+    } catch (e) {
+      throw Error("Tried to setSinkId " + sinkId + ", failed due to: " + e);
+    }
+  }
+
   _applyVolume() {
     const level = this.volume + this.trim;
     const linear = Math.pow(10, level / 20);
     if (linear < 1) {
       this.wavesurfer.setVolume(linear);
-      (this.wavesurfer as any).backend.gainNode.gain.value = 1;
+      if (!this.customOutput) {
+        (this.wavesurfer as any).backend.gainNode.gain.value = 1;
+      }
     } else {
       this.wavesurfer.setVolume(1);
-      (this.wavesurfer as any).backend.gainNode.gain.value = linear;
+      if (!this.customOutput) {
+        (this.wavesurfer as any).backend.gainNode.gain.value = linear;
+      }
```

Comment:
Where does volume get applied in customOutput mode?

Reply:
I did hide the options for customOutput, but it does works upto 100%, but it doesn't get any louder. (It's presumably a mapping to the volume option on the <audio> element)